d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancerState.java

```diff
@@ -967,16 +835,16 @@ void refreshTransportClientsPerService(ServiceProperties serviceProperties)
     if (uriProperties != null)
     {
       Set<URI> uris = uriProperties.Uris();
-      // clients-by-uri map may be edited later by UriPropertiesListener.handlePut
-      newTrackerClients = new ConcurrentHashMap<URI, TrackerClient>(
-          CollectionUtils.getMapInitialCapacity(uris.size(), 0.75f), 0.75f, 1);
-      long trackerClientInterval = getTrackerClientInterval (serviceProperties);
-      String errorStatusPattern = getErrorStatusPattern(serviceProperties);
+      newTrackerClients = new ConcurrentHashMap<>(CollectionUtils.getMapInitialCapacity(uris.size(), 0.75f), 0.75f, 1);
+
       for (URI uri : uris)
       {
-        TrackerClient trackerClient = buildTrackerClient(serviceName, uri, uriProperties.getPartitionDataMap(uri),
-                                                       config, clk, trackerClientInterval, errorStatusPattern,
-                                                       uriProperties.getUriSpecificProperties().get(uri));
+
+        TrackerClient trackerClient = TrackerClientFactory.createTrackerClient(uri,
+                                                                               uriProperties,
+                                                                               serviceProperties,
+                                                                               _serviceStrategies.get(serviceName).get(uri.getScheme()).getName(),
+                                                                               _serviceClients.get(serviceName).get(uri.getScheme()));
```

Comment:
I don't follow, can you clarify?

Reply:
_serviceStrategies.get(serviceName).get(uri.getScheme()) == newTrackerClient ?

Reply:
_serviceStrategies.get(serviceName).get(uri.getScheme()) == the service strategy for that scheme; if I understand you correctly.

Reply:
Can we replace _serviceStrategies.get(serviceName).get(uri.getScheme() with newTrackerClient?

Reply:
No since "Can we replace _serviceStrategies.get(serviceName).get(uri.getScheme()" doesn't return newTrackerClient? It returns an instance of LoadBalancerStrategy.